### PR TITLE
Add msbuild support

### LINKED ISSUE #8: No support for MsBuild variables
## Expected behaviour 

Process and evaluate `UserSecretsId` node to respect MsBuild properties.

It's possible to use MsBuild's well-defined variables as user secret values.
For example, using Visual Studio 2019, for a project `MyProject.csproj`

```xml
<UserSecretsId>$(MSBuildProjectName)</UserSecretsId>
```

`secrets.json` file will be created in `MyProject` folder rather than a guild.

## Actual behaviour

MsBuild value is treated as text and is used as-is, creating `secrets.json` in a folder named `$(MSBuildProjectName)`.

https://github.com/Witik/RiderUserSecrets/blob/27b11e601ab866cc45c733c5081374327c958888/src/eu/gillissen/rider/usersecrets/OpenUserSecretsAction.kt#L64-L68

### PR DESCRIPTION
**Pull request by maartenba**
Fixes https://github.com/Witik/RiderUserSecrets/issues/8

(built on top of previous PR)

### CODE REVIEW

File: src/main/kotlin/eu/gillissen/rider/usersecrets/OpenUserSecretsAction.kt [hunk 1]
```diff
@@ -2,75 +2,107 @@ package eu.gillissen.rider.usersecrets
...
+
+                val secretsId = msBuildProperties[UserSecretsIdMsBuildProperty] ?: return
+
+                ApplicationManager.getApplication().invokeLaterOnWriteThread {
+                    val secretsDirectoryRoot = getSecretsDirectoryRoot()
+                    val secretsDirectory = "$secretsDirectoryRoot${File.separatorChar}$secretsId"
+                    val secretsFile = File("$secretsDirectory${File.separatorChar}secrets.json")
+                    if (!secretsFile.exists()) {
+                        File(secretsDirectoryRoot).mkdirs()
+                        File(secretsDirectory).mkdirs()
```
Comment (Witik) [inline src/main/kotlin/eu/gillissen/rider/usersecrets/OpenUserSecretsAction.kt]: shouldn't just `File(secretsDirectory).mkdirs()` be sufficient here, as it creates all missing parent directories?

File: src/main/kotlin/eu/gillissen/rider/usersecrets/OpenUserSecretsAction.kt [hunk 2]
```diff
@@ -2,75 +2,107 @@ package eu.gillissen.rider.usersecrets
...
 
-        if (project == null || project.isDefault) {
-            actionEvent.presentation.isVisible = false
-            actionEvent.presentation.isEnabled = false
+        val project = CommonDataKeys.PROJECT.getData(actionEvent.dataContext) ?: return
+
+        if (project.isDefault) {
+            actionEvent.presentation.isEnabledAndVisible = false
             return
         }
```
Comment (Witik) [inline src/main/kotlin/eu/gillissen/rider/usersecrets/OpenUserSecretsAction.kt]: This deviates from the original behaviour where `isEnabledAndVisible` is set to `false` when `project == null`.

In the case that it's not necessary to set it to false, please update the other branches as well.

**Review by Witik — CHANGES_REQUESTED** [review]
Thanks for the PR, I learned a lot while looking through the code.

I made some suggestions, please look them over when you have the chance.

File: src/main/kotlin/eu/gillissen/rider/usersecrets/OpenUserSecretsAction.kt (hunk 1 above)
Comment (maartenba) [reply src/main/kotlin/eu/gillissen/rider/usersecrets/OpenUserSecretsAction.kt]: You're right, updated.

File: src/main/kotlin/eu/gillissen/rider/usersecrets/OpenUserSecretsAction.kt (hunk 2 above)
Comment (maartenba) [reply src/main/kotlin/eu/gillissen/rider/usersecrets/OpenUserSecretsAction.kt]: Good catch, updating.

**Review by Witik — APPROVED** [review]
LGTM, I will merge this PR if you can resolve its conflicts

**Comment by maartenba** [comment]
Sure thing, there you go :)

(also any chance for an updated plugin version on the marketplace? :-))

**Comment by Witik** [comment]
Thanks, I released the new version, but I'm not sure about the compatibility tags.

I think I released them for 2020.2 only, but it would probably work as a 2020.2+ release

**Pull request merged**